### PR TITLE
Fix blue background.

### LINKED ISSUE #2946: Toolstrip backcolor is off
All WPF toolstrips have a blue-ish background color that subtly clashes with the more grey backcolor of the containing docked panel.

Also, the dropdown menus on these toolstrips have a dark grey backcolor that less-subtly clashes with, well, everything around it:

image

This should be a relatively easy XAML fix.

### PR DESCRIPTION
**Pull request by IvenBach**
Closes #2946. F5 tested and UnitTested.

There appears to be some nasty hidden in here. I'll dig through to understand what's causing it. #3545  opened to bring attention to it.

### CODE REVIEW

**Comment by retailcoder** [comment]
I think this PR should address *all* toolwindow toolbars - they all have the same problem; fixing only one makes it too easy to forget about the others.

File: RetailCoder.VBE/UI/CodeExplorer/Commands/AddComponentCommand.cs [hunk 1]
```diff
@@ -52,20 +52,31 @@ private Declaration GetDeclaration(CodeExplorerItemViewModel node)
...
+            //{
+            //    return DefaultFolder;
+            //}
+
+            //if (node is ICodeExplorerDeclarationViewModel declarationNode)
+            //{
+            //    return string.IsNullOrEmpty(declarationNode.Declaration.CustomFolder)
+            //        ? DefaultFolder
+            //        : declarationNode.Declaration.CustomFolder.Replace("\"", string.Empty);
+            //}
```
Comment (Vogel612) [inline RetailCoder.VBE/UI/CodeExplorer/Commands/AddComponentCommand.cs]: commented code?

File: RetailCoder.VBE/UI/CodeExplorer/Commands/AddComponentCommand.cs [hunk 2]
```diff
@@ -52,20 +52,31 @@ private Declaration GetDeclaration(CodeExplorerItemViewModel node)
...
 
-            var declarationNode = node as ICodeExplorerDeclarationViewModel;
-            if (declarationNode != null)
+            //return ((CodeExplorerCustomFolderViewModel)node).FullPath;
```
Comment (Vogel612) [inline RetailCoder.VBE/UI/CodeExplorer/Commands/AddComponentCommand.cs]: commented out code?

File: RetailCoder.VBE/UI/ToDoItems/ToDoExplorerControl.xaml.cs [hunk 3]
```diff
@@ -6,15 +6,16 @@ namespace Rubberduck.UI.ToDoItems
     /// </summary>
     public partial class ToDoExplorerControl
     {
-        private ToDoExplorerViewModel ViewModel { get { return DataContext as ToDoExplorerViewModel; } }
+        //private ToDoExplorerViewModel ViewModel { get { return DataContext as ToDoExplorerViewModel; } }
```
Comment (Vogel612) [inline RetailCoder.VBE/UI/ToDoItems/ToDoExplorerControl.xaml.cs]: commented out code?

File: RetailCoder.VBE/UI/About/AboutControlViewModel.cs [hunk 4]
```diff
@@ -16,13 +16,7 @@ public AboutControlViewModel(IVersionCheck version)
...
         }
 
-        public string Version
-        {
-            get
-            {
-                return string.Format(RubberduckUI.Rubberduck_AboutBuild, _version.CurrentVersion);
-            }
-        }
+        public string Version => string.Format(RubberduckUI.Rubberduck_AboutBuild, _version.CurrentVersion);
```
Comment (Hosch250) [inline RetailCoder.VBE/UI/About/AboutControlViewModel.cs]: Nice.

File: RetailCoder.VBE/UI/UnitTesting/TestExplorerControl.xaml [hunk 5]
```diff
@@ -16,9 +16,13 @@
 
     <UserControl.Resources>
         <BitmapImage x:Key="SettingsImage" UriSource="../../Resources/gear.png" />
+
         
         <local:TestOutcomeImageSourceConverter x:Key="OutcomeIconConverter" />
+        
+        
         <local:TestResultToOutcomeTextConverter x:Key="OutcomeTextConverter" />
+
```
Comment (Vogel612) [inline RetailCoder.VBE/UI/UnitTesting/TestExplorerControl.xaml]: pretty excessive whitespace additions?

File: RetailCoder.VBE/UI/CodeExplorer/Commands/AddComponentCommand.cs [hunk 6]
```diff
@@ -52,20 +52,31 @@ private Declaration GetDeclaration(CodeExplorerItemViewModel node)
 
         private string GetFolder(CodeExplorerItemViewModel node)
         {
-            if (node == null)
-            {
-                return DefaultFolder;
-            }
+            //if (node == null)
```
Comment (Hosch250) [inline RetailCoder.VBE/UI/CodeExplorer/Commands/AddComponentCommand.cs]: Delete code. Don't comment it.

File: RetailCoder.VBE/UI/CodeExplorer/Commands/AddComponentCommand.cs [hunk 7]
```diff
@@ -52,20 +52,31 @@ private Declaration GetDeclaration(CodeExplorerItemViewModel node)
...
+            //    return string.IsNullOrEmpty(declarationNode.Declaration.CustomFolder)
+            //        ? DefaultFolder
+            //        : declarationNode.Declaration.CustomFolder.Replace("\"", string.Empty);
+            //}
 
-            var declarationNode = node as ICodeExplorerDeclarationViewModel;
-            if (declarationNode != null)
+            //return ((CodeExplorerCustomFolderViewModel)node).FullPath;
+
+            switch (node)
```
Comment (Hosch250) [inline RetailCoder.VBE/UI/CodeExplorer/Commands/AddComponentCommand.cs]: Nice use of pattern matching.

File: RetailCoder.VBE/UI/UnitTesting/TestExplorerViewModel.cs [hunk 8]
```diff
@@ -351,15 +355,15 @@ private void ExecuteCopyResultsCommand(object parameter)
 
             var aResults = Model.Tests.Select(test => test.ToArray()).ToArray();
 
-            var resource = "Rubberduck Test Results - {0}";
+            const string resource = "Rubberduck Test Results - {0}";
```
Comment (Vogel612) [inline RetailCoder.VBE/UI/UnitTesting/TestExplorerViewModel.cs]: mark for i18n cleanup?

File: RetailCoder.VBE/UI/Controls/DeclarationTypeToStringConverter.cs [hunk 9]
```diff
@@ -9,14 +9,13 @@ public class DeclarationTypeToStringConverter : IValueConverter
     {
         public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
         {
-            if (value is DeclarationType)
+            if (!(value is DeclarationType type))
```
Comment (Hosch250) [inline RetailCoder.VBE/UI/Controls/DeclarationTypeToStringConverter.cs]: Yuck. Don't invert conditions needlessly. It doesn't reduce nesting enough to talk about, for all R#'s shooting it's mouth off.

File: RetailCoder.VBE/UI/Controls/EmptyUIRefresh.xaml.cs [hunk 10]
```diff
@@ -1,17 +1,4 @@
...
-using System.Threading.Tasks;
-using System.Windows;
-using System.Windows.Controls;
-using System.Windows.Data;
-using System.Windows.Documents;
-using System.Windows.Input;
-using System.Windows.Media;
-using System.Windows.Media.Imaging;
-using System.Windows.Navigation;
-using System.Windows.Shapes;
```
Comment (Hosch250) [inline RetailCoder.VBE/UI/Controls/EmptyUIRefresh.xaml.cs]: Nice killing dead code.

File: RetailCoder.VBE/UI/Controls/MenuItemGroup.cs [hunk 11]
```diff
@@ -30,31 +30,34 @@ public static string GetGroupName(MenuItem element)
         private static void OnGroupNameChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
         {
             //Add an entry to the group name collection
-            var menuItem = d as MenuItem;
 
-            if (menuItem != null)
+            if (!(d is MenuItem menuItem))
```
Comment (Hosch250) [inline RetailCoder.VBE/UI/Controls/MenuItemGroup.cs]: Yuck.

File: RetailCoder.VBE/UI/Controls/NumberPicker.xaml.cs [hunk 12]
```diff
@@ -14,30 +14,17 @@ public partial class NumberPicker : IDataErrorInfo
 
         public int NumValue
         {
-            get
-            {
-                return (int)GetValue(NumValueProperty);
-            }
+            get => (int)GetValue(NumValueProperty);
```
Comment (Hosch250) [inline RetailCoder.VBE/UI/Controls/NumberPicker.xaml.cs]: Not a deal-breaker, but I would use braces for the get if you are going to use them for the set...

File: RetailCoder.VBE/UI/Inspections/InspectionImageSourceConverter.cs [hunk 13]
```diff
@@ -9,8 +9,7 @@ public class InspectionImageSourceConverter : IValueConverter
     {
         public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
         {
-            var inspection = value as IInspection;
-            if (inspection == null)
+            if (!(value is IInspection inspection))
```
Comment (Hosch250) [inline RetailCoder.VBE/UI/Inspections/InspectionImageSourceConverter.cs]: Yuck.

File: RetailCoder.VBE/UI/ToDoItems/ToDoExplorerControl.xaml.cs (hunk 3 above)
Comment (Hosch250) [inline RetailCoder.VBE/UI/ToDoItems/ToDoExplorerControl.xaml.cs]: Delete code. Don't comment it.

File: RetailCoder.VBE/UI/UnitTesting/TestExplorerControl.xaml [hunk 14]
```diff
@@ -16,9 +16,13 @@
 
     <UserControl.Resources>
         <BitmapImage x:Key="SettingsImage" UriSource="../../Resources/gear.png" />
+
         
         <local:TestOutcomeImageSourceConverter x:Key="OutcomeIconConverter" />
+
```
Comment (Hosch250) [inline RetailCoder.VBE/UI/UnitTesting/TestExplorerControl.xaml]: What's the deal with all this space?

File: RetailCoder.VBE/UI/UnitTesting/TestExplorerViewModel.cs (hunk 8 above)
Comment (Hosch250) [inline RetailCoder.VBE/UI/UnitTesting/TestExplorerViewModel.cs]: Nice.

File: RetailCoder.VBE/UI/UnitTesting/TestOutcomeImageSourceConverter.cs [hunk 15]
```diff
@@ -24,9 +24,9 @@ public class TestOutcomeImageSourceConverter : IValueConverter
 
         public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
         {
-            if (value.GetType() != typeof (TestOutcome))
+            if (value?.GetType() != typeof(TestOutcome))
             {
-                throw new ArgumentException("value must be a TestOutcome");
```
Comment (Hosch250) [inline RetailCoder.VBE/UI/UnitTesting/TestOutcomeImageSourceConverter.cs]: We threw for a reason... Only return `null` if the value passed in is `null`. (Unless Mat says the reason we threw isn't needed anymore.)

Comment (IvenBach) [reply RetailCoder.VBE/UI/UnitTesting/TestOutcomeImageSourceConverter.cs]: They appear to only be throwing when in the XAML design viewer. I agree that if they are there, it's for a reason. https://chat.stackexchange.com/transcript/message/41200243#41200243 was the suggestion Mug offered. If there's a way to keep them without throwing while in the designer that seems the best option.

Comment (Hosch250) [reply RetailCoder.VBE/UI/UnitTesting/TestOutcomeImageSourceConverter.cs]: OK, looks good then.

**Pull request merged**